core/trino-main/src/main/java/io/trino/dispatcher/DispatchManager.java

```diff
@@ -154,8 +154,6 @@ public ListenableFuture<Void> createQuery(QueryId queryId, Span querySpan, Slug
         DispatchQueryCreationFuture queryCreationFuture = new DispatchQueryCreationFuture();
         dispatchExecutor.execute(Context.current().wrap(() -> {
             Span span = tracer.spanBuilder("dispatch")
-                    .addLink(Span.current().getSpanContext())
-                    .setParent(Context.current().with(querySpan))
```

Comment:
I think this change is wrong. Dispatch is part of query processing and thus should retain the query span as the parent. We add the link here in order to track the client request that triggered the dispatch (this will be after the initial request which created the query).

With this change, the dispatch process will be completely detached from the query span.

Reply:
You're right! I'm not that familiar with the query execution flow, and I haven't noticed this happens in a separate request (`getStatus`) than creating the query span (`postStatement`). I reverted these changes.